File: ESAMI/_Svolti [OLD]/02-11-2017/java-aux/Driver.java
public class Driver {


    public static void main(String[] argv) {

    	BST b1 = new BST();
        b1.BST_insert(3);
        b1.BST_insert(12);
        b1.BST_insert(7);
        b1.BST_insert(5);
        b1.BST_insert(1);
        b1.BST_insert(7);
        b1.BST_insert(15);
        b1.BST_insert(1);
        b1.BST_insert(8);
        b1.BST_insert(7);
        b1.BST_insert(12);

    	BST b2 = new BST();
    	
        BinNode root = b2.BST_insert(13);
        if (root!=null){
            BinNode l1 = root.setLeft(new BinNode(5));
            BinNode r1 = root.setRight(new BinNode(20));
            BinNode l2 = l1.setLeft(new BinNode(3));
            BinNode r2 = l1.setRight(new BinNode(17));
            BinNode l3 = r1.setRight(new BinNode(25));
        }


        System.out.println("\t Primo Albero: ");
        b1.BST_print();

        System.out.print("Test BST: ");
        int BSTresult=b1.isBST();
        if (BSTresult == -1){
            System.out.println("Funzione non implementata");     
        }else if (BSTresult == 0){
            System.out.println("Albero non BST");     
        }else{
            System.out.println("Albero BST");     
        } 

        System.out.print("Test Balanced: ");
        int balresult=b1.isBalanced();
        if (balresult == -1){
            System.out.println("Funzione non implementata");     
        }else if (balresult == 0){
            System.out.println("Albero non bilanciato");     
        }else{
            System.out.println("Albero bilanciato");     
        } 

        System.out.print("Test AVL: ");
        int AVLresult=b1.isAVL();
        if (AVLresult == -1){
            System.out.println("Funzione non implementata");     
        }else if (AVLresult == 0){
            System.out.println("Albero non AVL");     
        }else{
            System.out.println("Albero AVL");     
        } 

        System.out.println("-------------------");
        System.out.println("\t Secondo Albero: ");
        b2.BST_print();

        System.out.print("Test BST: ");
        BSTresult=b2.isBST();
        if (BSTresult == -1){
            System.out.println("Funzione non implementata");     
        }else if (BSTresult == 0){
            System.out.println("Albero non BST");     
        }else{
            System.out.println("Albero BST");     
        } 

        System.out.print("Test Balanced: ");
        balresult=b2.isBalanced();
        if (balresult == -1){
            System.out.println("Funzione non implementata");     
        }else if (balresult == 0){
            System.out.println("Albero non bilanciato");     
        }else{
            System.out.println("Albero bilanciato");     
        } 

        System.out.print("Test AVL: ");
        AVLresult=b2.isAVL();
        if (AVLresult == -1){
            System.out.println("Funzione non implementata");     
        }else if (AVLresult == 0){
            System.out.println("Albero non AVL");     
        }else{
            System.out.println("Albero AVL");     
        } 



    }
}

File: ESAMI/Svolti_[OLD]/Storage/02-11-2017/c-aux/driver.c
#include <stdio.h>
#include <stdlib.h>
#include "bst.h"




int main(void) {
    BST *b1 = newBST();
    BST_insert(b1, 3);
    BST_insert(b1, 12);
    BST_insert(b1, 7);
    BST_insert(b1, 5);
    BST_insert(b1, 1);
    BST_insert(b1, 7);
    BST_insert(b1, 15);
    BST_insert(b1, 1);
    BST_insert(b1, 8);
    BST_insert(b1, 7);
    BST_insert(b1, 12);

    BST *b2 = newBST();
    	
    BinNode *root, *l1, *l2, *l3, *r1, *r2;
    root = BST_insert(b2, 13);
    root->left = l1 = newBinNode(5);
    root->right = r1 = newBinNode(20);
    l1->left = l2 = newBinNode(3);
    l1->right = r2 = newBinNode(17);
    r1->right = l3 = newBinNode(25);
        
    printf("\t Primo Albero: \n");
    BST_print(b1);

    printf("Test BST: ");
    int BSTresult = isBST(b1);
    if (BSTresult == -1) printf("Funzione non implementata\n");     
    else if (BSTresult == 0) printf("Albero non BST\n");     
    else printf("Albero BST\n");     
    
    printf("Test Balanced: ");
    int balresult = isBalanced(b1);
    if (balresult == -1) printf("Funzione non implementata\n");     
    else if (balresult == 0) printf("Albero non bilanciato\n");     
    else printf("Albero bilanciato\n");     

    printf("Test AVL: ");
    int AVLresult = isAVL(b1);
    if (AVLresult == -1) printf("Funzione non implementata\n");     
    else if (AVLresult == 0) printf("Albero non AVL\n");     
    else printf("Albero AVL\n");     
    

    printf("-------------------\n");
    printf("\t Secondo Albero: \n");
    BST_print(b2);

    printf("Test BST: ");
    BSTresult = isBST(b2);
    if (BSTresult == -1) printf("Funzione non implementata\n");     
    else if (BSTresult == 0) printf("Albero non BST\n");     
    else printf("Albero BST\n");     
    
    printf("Test Balanced: ");
    balresult = isBalanced(b2);
    if (balresult == -1) printf("Funzione non implementata\n");     
    else if (balresult == 0) printf("Albero non bilanciato\n");     
    else printf("Albero bilanciato\n");     
    

    printf("Test AVL: ");
    AVLresult = isAVL(b2);
    if (AVLresult == -1) printf("Funzione non implementata\n");     
    else if (AVLresult == 0) printf("Albero non AVL\n");     
    else printf("Albero AVL\n");
    
    return 0;
}
